stellargraph/core/graph.py

```diff
@@ -944,6 +944,75 @@ def to_adjacency_matrix(self, nodes: Optional[Iterable] = None, weighted=False):
         adj.sum_duplicates()
         return adj
 
+    def subgraph(self, nodes):
+        """
+        Compute the node-induced subgraph implied by ``nodes``.
+
+        Args:
+            nodes (iterable): The nodes in the subgraph.
+
+        Returns:
+            A :class:`StellarGraph` or :class:`StellarDiGraph` instance containing only the nodes in
+            ``nodes``, and any edges between them in ``self``. It contains the same node & edge
+            types, node features and edge weights as in ``self``.
+        """
+
+        node_ilocs = self._nodes.ids.to_iloc(nodes, strict=True)
+        node_types = self._nodes.type_of_iloc(node_ilocs)
+        node_type_to_ilocs = pd.Series(node_ilocs, index=node_types).groupby(level=0)
+
+        node_frames = {
+            type_name: pd.DataFrame(
+                self._nodes.features(type_name, ilocs),
+                index=self._nodes.ids.from_iloc(ilocs),
+            )
+            for type_name, ilocs in node_type_to_ilocs
+        }
+
+        # FIXME: this is O(edges in graph) but could potentially be optimised to O(edges incident in
```

Comment:
can you create an issue for this (if you haven't already!) and document it here?

Reply:
#985